src/direc.rs

```diff
@@ -164,6 +180,18 @@ pub struct DirecVector<T> {
     pub direction: Direction,
 }
 
+/// A scalar multiplication for a vector.
+impl<T> std::ops::Mul<T> for &DirecVector<T>
+where
+    T: std::ops::Mul<Output = T> + Copy,
```

Comment:
I think `Copy` is too restrictive. You could replace it with a `T: Clone` constraint instead.

Reply:
I absolutely agree; even already made the change locally (AFAIU for copy-types `clone` just calls the same copy method). Apparently this modification would be better from theoretically (so why not do it better); and still I try to justify it with some example...

After some thinking let me put it following way. Why `DirecVector` `magnitude` is *so generic*? It it just because you can,  X) or any reasons behind it? Because I feel like I would make it `Integer` and I really would like to nurture some better intuition in me than I have now. X)